client/my-sites/hosting/github/github-connect-card/index.tsx

```diff
@@ -1,16 +1,106 @@
 export const GithubConnectCard = () => {
-	const { __ } = useI18n();
+	const siteId = useSelector( getSelectedSiteId );
+
+	const [ selectedRepo, setSelectedRepo ] = useState< Repo >();
+	const [ selectedBranch, setSelectedBranch ] = useState< RepoBranch >();
+	const { data: repos, isLoading: isLoadingRepos } = useGithubRepos( siteId );
+	const { data: branches, isLoading: isLoadingBranches } = useGithubBranches(
+		siteId,
+		selectedRepo?.name
+	);
+
+	useEffect( () => {
+		if ( repos ) {
+			setSelectedRepo( selectedRepo || repos[ 0 ] );
+		}
+		if ( branches ) {
+			setSelectedBranch( branches[ 0 ] );
+		}
+	}, [ repos, branches, selectedRepo ] );
+
+	const showSpinner = ! repos && ! branches && ( isLoadingRepos || isLoadingBranches );
+
+	const handleRepoSelect = ( repoName: string ) => {
+		const selectedRepo = repos?.find( ( repo ) => repo.full_name === repoName );
+		setSelectedRepo( selectedRepo );
+	};
+
+	const handleBranchSelect = ( branchName: string ) => {
+		const selectedBranch = branches?.find( ( branch ) => branch.name === branchName );
+		setSelectedBranch( selectedBranch );
+	};
+
 	return (
-		<Card className="github-hosting-card">
-			<img className="github-hosting-icon" src={ iconGitHub } alt="" />
+		<Card className="connect-branch-card">
+			<SocialLogo className="material-icon" icon="github" size={ 32 } />
 			<CardHeading>{ __( 'Connect Branch' ) }</CardHeading>
+			<div>
+				<p>{ __( 'Changes pushed to connected branch will be deployed automatically' ) }</p>
+				{ ! showSpinner ? (
+					<FormFieldset className="connect-branch__fields">
+						<FormSelect
+							id="repositories"
+							className="connect-branch__field"
+							onChange={ ( event ) => handleRepoSelect( event.currentTarget.value ) }
+							value={ selectedRepo?.full_name }
+						>
+							{ repos?.map( ( repo ) => (
+								<option value={ repo.full_name } key={ repo.name }>
+									{ repo.full_name }
+								</option>
+							) ) }
+						</FormSelect>
+						<FormSelect
+							id="branches"
+							className="connect-branch__field"
+							onChange={ ( event ) => handleBranchSelect( event.currentTarget.value ) }
+							value={ selectedBranch?.name }
+						>
+							{ branches?.map( ( branch ) => (
+								<option value={ branch.name } key={ branch.name }>
+									{ branch.name }
+								</option>
+							) ) }
+						</FormSelect>
+						<Button
+							primary
+							busy={ isLoadingBranches }
+							className="connect-branch__field"
+							disabled={ isLoadingBranches }
+						>
+							<span>{ __( 'Connect' ) }</span>
+						</Button>
+					</FormFieldset>
+				) : (
+					<Spinner />
+				) }
+				<div>
+					<span>
+						<strong>{ __( 'Root Directory' ) }</strong>
+						<small>
+							{ __(
+								'(you can specify a root directory and only changes in that directory will be deployed)'
+							) }
+						</small>
+					</span>
+					<FormTextInput id="root_path" placeholder="/wp-content" />
```

Comment:
I'm not sure whether this placeholder should be using `__()` 🙃 Probably not right?
I do think the input should have a `dir="ltr"` attribute though so that even in RTL languages the browser knows to render the contents as a LTR string. This is how we present domains in the Calypso UI, so I think it makes sense for file paths too.

Reply:
Yes I think `dir="ltr"` is what we want.

Without the attribute it's unable to render an absolute path correctly.
<img width="611" alt="CleanShot 2023-02-07 at 21 27 09@2x" src="https://user-images.githubusercontent.com/1500769/217191718-9f53c32f-8cc3-42e9-bf81-a527d4fda282.png">

This is what it's like with the attribute.
<img width="592" alt="CleanShot 2023-02-07 at 21 28 35@2x" src="https://user-images.githubusercontent.com/1500769/217191990-cb7efedd-3673-4455-aa21-1532fb9f5e00.png">

Reply:
Thanks didn't know about the dir. 

> I'm not sure whether this placeholder should be using __() 🙃 Probably not right?

I think it should remain as is since it's a folder path that's language agnostic. Might be difficult to say how translators should represent a folder path.